src/app/terminal.js

```diff
@@ -0,0 +1,301 @@
+    function hover (event) { event.currentTarget.classList.toggle(css.hover) }
+    function minimize (event) {
+      event.preventDefault()
+      event.stopPropagation()
+      var classList = self._view.icon.classList
+      classList.toggle('fa-angle-double-down')
+      classList.toggle('fa-angle-double-up')
+      self.event.trigger('resize', [])
+    }
+    // ----------------- resizeable ui ---------------
+    function mousedown (event) {
+      event.preventDefault()
+      if (event.which === 1) {
+        moveGhostbar(event)
+        document.body.appendChild(ghostbar)
+        document.addEventListener('mousemove', moveGhostbar)
+        document.addEventListener('mouseup', removeGhostbar)
+        document.addEventListener('keydown', cancelGhostbar)
+      }
+    }
+    function cancelGhostbar (event) {
+      if (event.keyCode === 27) {
+        document.body.removeChild(ghostbar)
+        document.removeEventListener('mousemove', moveGhostbar)
+        document.removeEventListener('mouseup', removeGhostbar)
+        document.removeEventListener('keydown', cancelGhostbar)
+      }
+    }
+    function moveGhostbar (event) { // @NOTE HORIZONTAL ghostbar
+      ghostbar.style.top = self._api.getPosition(event) + 'px'
+    }
+    function removeGhostbar (event) {
+      if (self._view.icon.classList.contains('fa-angle-double-up')) {
+        self._view.icon.classList.toggle('fa-angle-double-down')
+        self._view.icon.classList.toggle('fa-angle-double-up')
+      }
+      document.body.removeChild(ghostbar)
+      document.removeEventListener('mousemove', moveGhostbar)
+      document.removeEventListener('mouseup', removeGhostbar)
+      document.removeEventListener('keydown', cancelGhostbar)
+      self.event.trigger('resize', [self._api.getPosition(event)])
+    }
+
+    return self._view.panel
+    function change (event) {
+      if (event.which === 13) {
+        if (event.ctrlKey) { // <ctrl+enter>
+          // console.log('enter')
+          self._view.input.appendChild(document.createElement('br'))
+          self.scroll2bottom()
+        } else {
+          event.preventDefault()
+        }
+      }
+      console.log(event.which, event.keyCode)
+      // console.log('----------------')
+      // console.log(event.key) // Control
+      // console.log('-----')
+      // console.log(event.code) // ControlLeft
+      // console.log('-----------')
+      // console.log('event.altKey ' + event.altKey)
+      // console.log('event.ctrlKey ' + event.ctrlKey)
+      // console.log('event.metaKey ' + event.metaKey)
+      // console.log('event.shiftKey ' + event.shiftKey)
+      // console.log(event.location)
+      // console.log('event.location')
+      // console.log(event.returnValue)
+      // console.log('event.returnValue')
+      // console.log(event.type)
+      // console.log('event.type')
+      // console.log('-----')
+      // console.log(event)
+    }
+  }
+  log (obj) {
+    var self = this
```

Comment:
we really need this to be addressed before merging. cause the `txlistener` needs to log a node element (containing the action to start debugging) and not only text.